controllers/logging/elasticsearch_controller.go

```diff
@@ -49,8 +51,14 @@ func (r *ElasticsearchReconciler) Reconcile(request ctrl.Request) (ctrl.Result,
 	}
 
 	if cluster.Spec.ManagementState == loggingv1.ManagementStateUnmanaged {
+		// Cluster state changes from Managed -> Unmanaged, so set "unmanaged" as 1 and set "managed" as 0.
+		metrics.SetEsClusterManagementStateManaged(0)
+		metrics.SetEsClusterManagementStateUnmanaged(1)
 		return ctrl.Result{}, nil
 	}
+	// Cluster state changes from Unmanaged -> Managed, so set "managed" as 1 and set "unmanaged" as 0.
+	metrics.SetEsClusterManagementStateManaged(1)
+	metrics.SetEsClusterManagementStateUnmanaged(0)
```

Comment:
nit. You could have hidden this numbers behind the helper functions, e.g. `metrics.SetEsClusterManagementStateManaged` is doing the `SetManaged(1)` and `SetUnmanaged(0)` in the background.